src/implementation/c/index.ts

```diff
@@ -35,6 +35,24 @@ export class CCompiler {
     out.push('#include <stdlib.h>');
     out.push('#include <stdint.h>');
     out.push('#include <string.h>');
+    out.push('');
+
+    // NOTE: Inspired by https://github.com/h2o/picohttpparser
+    out.push('#ifdef __SSE4_2__');
```

Comment:
Are you defining this somewhere? MSVC doesn't define this like gcc/clang do.

Reply:
Oh no. Do they define anything at all?

Reply:
No, it's annoying, but nothing between SSE2 and AVX is indicated by a macro. MSVC does make it easier to use `cpuid` than GCC if you can deal with a runtime check (set a function pointer once to avoid the overhead), and MSVC will happily emit SSE4.2 insns from their intrinsics with the default build config (I think... I'm not 100% certain if that's true for 32-bit builds but it is for 64-bit -- I can check if needed). https://docs.microsoft.com/en-us/cpp/intrinsics/cpuid-cpuidex?view=vs-2019

Reply:
I suppose we could remove MSVC support for now, until some resolution will be attained. Thank you very much for bringing this up!

Reply:
No problem. I can help with a solution sometime. I don't remember what Node's default build config allows (probably only SSE2?) but multiple-dispatch might be a good idea for all platforms.

Reply:
:+1: That would be great, if you have time. Please hit me up here, on twitter, or over email with any questions about the project.

Thanks!

Reply:
Will do, thanks -- will be a few days until I can get to it (and anyone else should feel free to do it if they get to it before me).

Reply:
I was thinking about it since yesterday. Given that there is no `__SSE4_2__` define on Windows, having code guarded by it should not cause any problems either. I'm reverting yesterday's change with that in mind, and adding a TODO comment.